src/Npgsql/ConnectorPool.cs

```diff
@@ -210,14 +208,7 @@ internal async ValueTask<NpgsqlConnector> AllocateLong(NpgsqlConnection conn, Np
                 if (state.Total < _max)
                 {
                     // We're under the pool's max capacity, try to "allocate" a slot for a new physical connection.
-                    newState.Busy++;
-                    CheckInvariants(newState);
-                    if (Interlocked.CompareExchange(ref State.All, newState.All, state.All) != state.All)
-                    {
-                        // Our attempt to increment the busy count failed, Loop again and retry.
-                        continue;
-                    }
-
+                    Interlocked.Increment(ref State.Busy);
```

Comment:
You may have already explained this in the dev-list, but what keeps this from racing (i.e. incrementing beyond max)?

Reply:
Good catch, I'll add a check after it that checks if our local state.Busy + return value of increment matches up to something below _max, otherwise back out. 

Changing this from a pessimistic implementation to a bit more optimistic.

Reply:
Yeah that just won't work, I'll back out of this part of the change.
We're actually really not depending much on Busy here but on the fact Total (comprised of Idle and Busy) stays under _max. CompareExchange is just a better fit for those dependent changes.